packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -149,7 +148,7 @@ export abstract class HandlebarsNodeVisitors extends Parser {
         break;
 
       default:
-        throw new Error(`Using a Handlebars comment when in the \`${tokenizer.state}\` state is not supported: "${comment.value}" on line ${loc.start.line}:${loc.start.column}`);
+        throw new SyntaxError(`Using a Handlebars comment when in the \`${tokenizer.state}\` state is not supported: "${comment.value}" on line ${loc.start.line}:${loc.start.column}`, rawComment.loc);
```

Comment:
I think we should; it makes the error message more useful and it would be weird to make it less useful imo